bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2StateDescriptionProvider.java

```diff
@@ -41,4 +56,43 @@ public Clip2StateDescriptionProvider(final @Reference EventPublisher eventPublis
         this.itemChannelLinkRegistry = itemChannelLinkRegistry;
         this.channelTypeI18nLocalizationService = channelTypeI18nLocalizationService;
     }
+
+    @Override
+    public @Nullable StateDescription getStateDescription(Channel channel, @Nullable StateDescription original,
+            @Nullable Locale locale) {
+        StateDescriptionFragment stateDescriptionFragment = stateDescriptionFragments.get(channel.getUID());
+        if (stateDescriptionFragment != null) {
+            StateDescriptionFragmentBuilder builder = original == null ? StateDescriptionFragmentBuilder.create()
+                    : StateDescriptionFragmentBuilder.create(original);
+            String pattern = original != null ? original.getPattern() : null;
+            pattern = pattern != null ? pattern : "%.0f K";
+            builder.withPattern(pattern);
+            BigDecimal minimum = stateDescriptionFragment.getMinimum();
+            if (minimum != null) {
+                builder.withMinimum(minimum);
+            }
+            BigDecimal maximum = stateDescriptionFragment.getMaximum();
+            if (maximum != null) {
+                builder.withMaximum(maximum);
+            }
+            return builder.build().toStateDescription();
+        }
+        return super.getStateDescription(channel, original, locale);
+    }
+
+    /**
+     * Set the state description minimum and maximum values for the given channel UID
+     */
+    public void setMinMax(ChannelUID channelUID, long min, long max) {
+        StateDescriptionFragment oldStateDescriptionFragment = stateDescriptionFragments.get(channelUID);
+        StateDescriptionFragment newStateDescriptionFragment = StateDescriptionFragmentBuilder.create()
+                .withMinimum(BigDecimal.valueOf(min)).withMaximum(BigDecimal.valueOf(max)).build();
+        if (!newStateDescriptionFragment.equals(oldStateDescriptionFragment)) {
+            stateDescriptionFragments.put(channelUID, newStateDescriptionFragment);
+            ItemChannelLinkRegistry itemChannelLinkRegistry = this.itemChannelLinkRegistry;
+            postEvent(ThingEventFactory.createChannelDescriptionChangedEvent(channelUID,
+                    itemChannelLinkRegistry != null ? itemChannelLinkRegistry.getLinkedItemNames(channelUID) : Set.of(),
+                    newStateDescriptionFragment, oldStateDescriptionFragment));
+        }
```

Comment:
Is it necessary to post such event ?
What would be the result exactly ?
As your fragment has only min and max, no problem to not have any pattern ?

Reply:
I copied the code from another dynamic state provider. So I don't really know the impact (at least until your PRs are merged). However my guess is that it would cause the UI to refresh if the min/max range of a lamp would change. In other words perhaps your dynamic gradient might adjust itself to match the lamp..

Reply:
> no problem to not have any pattern ?

It uses any existing pattern if present and `%.0f K` if not.

Reply:
That's strange, I already implemented dynamic state description providers but I never used such mechanism.

Reply:
The code is copied from the super class which is part of OH so I suppose there is some reason for it.

Reply:
I generally used the method setStateOptions from the parent class which is doing something very similar. But there is nothing in this parent class for min/max, so now I understand.

Reply:
I found in our codebase 3 bindings calling .createChannelDescriptionChangedEvent. One is already  in the hue binding: class HueStateDescriptionProvider. You don't want to use it ?

Reply:
The HueStateDescriptionProvider is for API v1 thing handlers, and the Clip2StateDescriptionProvider is for API v2 thing handlers. I know the code is similar, but I wanted to keep the code base of the two API versions apart.

Reply:
Ok.
I just need to understand if this is a problem that you post a changed event for state description with min and max but without any pattern.
It is different to other bindings.

Reply:
Hmm. The getStateDescription method above does add the pattern which may be copied from 'original'. However, as you mention, that pattern is not available in the post event.

Reply:
> However, as you mention, that pattern is not available in the post event.

Yes and I don't know the impact of that.
I don't know but maybe it is finally interpreted by Main UI for example.

Reply:
That is what I was thinking. Unfortunately I don't know any way to do it less indirectly.

Reply:
Basic UI is listening for ChannelDescriptionChangedEvent but it just reloads the page whatever the content of the event.
I don't know what is doing Main UI.

Reply:
> That is what I was thinking. Unfortunately I don't know any way to do it less indirectly.

You could simply build a fragment with min max and pattern. It even simplifies the other method as you haven't to consider the pattern from the original.

Reply:
Ok. I will try that.

Reply:
Take that as an example (from the same binding):
https://github.com/openhab/openhab-addons/blob/abe819978feec579f26e94d9f4f4cff9b7ba3b76/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueLightHandler.java#L219